### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by K0ppai**
Dear reviewers,
We've worked on the following in this pull request,
- Install CanCanCan in the project.
- Add a `role` column to the users table.
- Add the following authorizations
  - A user can delete a post if it is theirs or if they have an admin role. 
  - A user can delete a comment if it is theirs or if they have an admin role.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    return unless user.present?
+
+    can :manage, Post, author_id: user.id
+    can :manage, Comment, user_id: user.id
```
Comment (DammyShittu) [inline app/models/ability.rb:9]: - [ ] The requirement is that only post owners or admins can manage posts or comments. Regular users who are neither should still be able to read posts, even if they can't delete them. Kindly add the following line here.

  ```
  can :read, :all
  ```
  
  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @K0ppai and @yemidada,

## Highlights
✔️ Custom `admin?` method in User model

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @K0ppai and @yemidada, 

## Highlights!
- Good code organization and working authorization ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by K0ppai** [comment]
Thank you for the reviews @DammyShittu, @julie-ify. 😁

**Pull request merged**